### PR TITLE
Migrate checkstyle to 8.12

### PR DESCRIPTION
**Pull request by Beacze**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -160,7 +160,14 @@
                     <plugin>
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-checkstyle-plugin</artifactId>
-                        <version>2.17</version>
+                        <version>3.0.0</version>
+                        <dependencies>
+                            <dependency>
+                                <groupId>com.puppycrawl.tools</groupId>
```
Comment (Kathrin-Huber) [inline pom.xml]: Why did you need to add a dependency here, which we did not need before?

Comment (Beacze) [reply pom.xml]: Because:

_This version of the plugin uses Checkstyle 6.18 by default and requires Java 7. But you can upgrade the version used at runtime._
Source: https://maven.apache.org/plugins/maven-checkstyle-plugin/
